openprescribing/data/fetchers/prescribing.py

```diff
@@ -1,2 +1,100 @@
+import datetime
+import logging
+import re
+import tempfile
+from pathlib import Path
+
+from openprescribing.data.utils.api_session import APISession
+from openprescribing.data.utils.csv_to_parquet import csv_to_parquet
+from openprescribing.data.utils.filename_utils import (
+    get_latest_files_by_date,
+    get_temp_filename_for,
+)
+from openprescribing.data.utils.zipfile_utils import extract_file_from_zip_archive
+
+
+log = logging.getLogger(__name__)
+
+
 def fetch(directory):
-    assert False
+    dataset_dir = directory / "prescribing"
+    existing_files = dataset_dir.glob("*")
+
+    api = APISession(
+        "https://opendata.nhsbsa.net/api/3/",
+        # We start with DEBUG level logging while we're checking if there's anything new
+        # to fetch
+        log=log.debug,
+    )
+    response = api.get(
+        "action/package_show",
+        params={"id": "english-prescribing-dataset-epd-with-snomed-code"},
+    )
+    response_data = response.json()
+
+    items_to_fetch = get_items_to_fetch(existing_files, response_data)
+
+    for item_id, output_filename in items_to_fetch:
+        # Now we're downloading new files we switch to INFO level logging
+        api.log = log.info
```

Comment:
This belongs outside the loop.

Reply:
Yes, I know. My original thinking was that we don't actually know that we _are_ downloading new files until we're inside the loop and I didn't want to add another conditional outside the loop. However now I think about it we can harmlessly set it outside the loop because we won't issue any more requests if there's nothing to download.